market/ipni/ipni-provider/ipni-provider.go

```diff
@@ -140,17 +159,23 @@ func NewProvider(d *deps.Deps) (*Provider, error) {
 		httpServerAddresses[i] = addr
 	}
 
+	prCache := ttlcache.NewCache()
+	_ = prCache.SetTTL(time.Minute * 10)
+	prCache.SetCacheSizeLimit(MaxCachedReaders)
```

Comment:
Maybe we should have this wrapped into a single, cached, sector reader utility in Deps? It's one more layer, but this code will likely get duplicated in other retrieval spots